### PR TITLE
fix: update FilterSearchType type definition

### LINKED ISSUE #37584: [TypeScript] FilterSearchType should use RecordType instead of {} for record
### Reproduction link

[https://github.com/ant-design/ant-design/blob/master/components/table/interface.tsx#L75](https://github.com/ant-design/ant-design/blob/master/components/table/interface.tsx#L75)

### Steps to reproduce

TypeScript problem

### What is expected?

```ts
export type FilterSearchType<RecordType> = boolean | ((input: string, record: RecordType) => boolean);
// ...
export interface ColumnType<RecordType> extends Omit<RcColumnType<RecordType>, 'title'> {
  // ...
  filterSearch?: FilterSearchType<RecordType>;
  // ...
}
```

### What is actually happening?

```ts
export type FilterSearchType = boolean | ((input: string, record: {}) => boolean);
```

| Environment | Info |
| --- | --- |
| antd | 4.23.1 |
| React | not related |
| System | not related |
| Browser | not related |

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [X] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

- fix: #37584 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    update `FilterSearchType`  definition     |
| 🇨🇳 Chinese |    更新 FilterSearchType 类型描述     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

**Comment by zation** [comment]
我感觉我之前的理解是错误的，FilterSearchType 里面的 record 是不是应该是 ColumnFilterItem？因为 FilterSearch 是 search 的 filters

**Comment by kiner-tang** [comment]
> 我感觉我之前的理解是错误的，FilterSearchType 里面的 record 是不是应该是 ColumnFilterItem？因为 FilterSearch 是 search 的 filters

image
不仅仅是`ColumnFilterItem `,目前还有这种类型的使用

File: components/config-provider/__tests__/__snapshots__/components.test.tsx.snap [hunk 1]
```diff
@@ -26508,288 +26508,43 @@ exports[`ConfigProvider components Statistic prefixCls 1`] = `
...
-        </div>
-        <div
-          class="config-steps-item-description"
-        >
-          Little Light
-        </div>
-      </div>
-    </div>
-  </div>
-</div>
```
Comment (afc163) [inline components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:26546]: 不应该修改这些

Comment (kiner-tang) [reply components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:26546]: 这个确实没有修改，但`yarn test:update`时更新了，我合并一下最新`master`再`update`看看

Comment (kiner-tang) [reply components/config-provider/__tests__/__snapshots__/components.test.tsx.snap:26546]: 找到原因了，切换分支后，没有重新安装依赖导致的

**Comment by afc163** [comment]
加个 typing 的用例？

**Comment by kiner-tang** [comment]
> 加个 ts type 的用例？

好的

**Comment by afc163** [comment]
看下 ci 报错

**Comment by kiner-tang** [comment]
> 看下 ci 报错
done

**Comment by afc163** [comment]
还有问题

**Pull request merged**